### PR TITLE
BUG 1826017: Switch to Go errors instead of github.com/pkg/errors

### PR DESCRIPTION
**Pull request by JoelSpeed**
Replace all `errors.Wrap(f)` with `fmt.Errorf` to drop the dependency on `github.com/pkg/errors`.

This ensures that the vSphere provider is compatible with the changes introduced in #559

### CODE REVIEW

File: pkg/controller/machineset/controller.go [hunk 1]
```diff
@@ -26,7 +27,6 @@ import (
 
 	machinev1beta1 "github.com/openshift/machine-api-operator/pkg/apis/machine/v1beta1"
 	"github.com/openshift/machine-api-operator/pkg/util"
```
Comment (enxebre) [inline pkg/controller/machineset/controller.go:29]: Seems the machineSet changes are for consistency but not really related to the BZ referenced, they probably deserve its own commit?

Comment (JoelSpeed) [reply pkg/controller/machineset/controller.go:29]: I've split the commit into two, one for vSphere and one for MachineSet

**Comment by enxebre** [comment]
We should go and put a counter part PR upstream for this.
/lgtm
